pithos/pithos.py

```diff
@@ -910,6 +918,58 @@ def station_changed(self, station, reconnecting=False):
         self.stations_popover.select_station(station)
         self.emit('station-changed', station)
 
+    def station_added(self, station, user_data):
+        music_type, description = user_data
+        for existing_station in self.stations_model:
+            if existing_station[0].id == station.id:
+                self.station_already_exists(existing_station[0], description, music_type, self)
+                return
+        # We shouldn't actually add the station to the pandora stations list
+        # until we know it's not a duplicate.
+        self.pandora.stations.append(station)
+        self.stations_model.insert_with_valuesv(0, (0, 1, 2), (station, station.name, 0))
+        self.emit('station-added', station)
+        self.station_changed(station)
+
+    def station_already_exists(self, station, description, music_type, parent):
+        def on_response(dialog, response):
+            if response == Gtk.ResponseType.YES:
+                self.station_changed(station)
+            dialog.destroy()
+
+        sub_title = _('Pandora does not permit multiple stations with the same seed.')
+
+        if music_type == 'song':
+            seed = _('Song Seed:')
+        elif music_type == 'artist':
+            seed = _('Artist Seed:')
+        else:
+            seed = _('Genre Seed:')
+
+        if station is self.current_station:
+            button_type = Gtk.ButtonsType.OK
+            message = _(', the Station you are currently listening to already contains the')
+            message = '{}\n"{}"{} {} {}'.format(sub_title, station.name, message, seed, description)
+
+        else:
+            button_type = Gtk.ButtonsType.YES_NO
+            a = _('Your Station')
+            b = _('already contains the')
```

Comment:
I'm not sure how translations work. There are several variables that shouldn't be translated in the middle of bits that should be translated.